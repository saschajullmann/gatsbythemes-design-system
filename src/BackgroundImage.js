import styled from '@emotion/styled'
import PropTypes from 'prop-types'
import { width } from 'styled-system'
import theme from './theme'

const image = props =>
  props.image ? { backgroundImage: `url(${props.image})` } : null

const height = props => (props.height ? { height: props.height } : null)

const BackgroundImage = styled.div`
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
  background-color: ${props => props.theme.colors.gray};
  ${image} ${height} ${width};
`
const numberStringOrArray = PropTypes.oneOfType([
  PropTypes.number,
  PropTypes.string,
  PropTypes.array
])

BackgroundImage.propTypes = {
  /** background-image url */
  image: PropTypes.string,
  width: numberStringOrArray
}

BackgroundImage.defaultProps = {
  theme: theme
}

BackgroundImage.displayName = 'BackgroundImage'

export default BackgroundImage
