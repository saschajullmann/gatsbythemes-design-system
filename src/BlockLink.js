import styled from '@emotion/styled'
import Link from './Link'

const BlockLink = styled(Link)`
  display: block;
  color: inherit;
  text-decoration: none;
`

BlockLink.displayName = 'BlockLink'

export default BlockLink
