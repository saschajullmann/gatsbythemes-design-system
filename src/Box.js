import styled from '@emotion/styled'
import { space, width, color, textAlign } from 'styled-system'
import PropTypes from 'prop-types'
import theme from './theme'

const Box = styled.div`
  ${space} ${width} ${color} ${textAlign}
`

Box.displayName = 'Box'

Box.defaultProps = {
  theme: theme
}

Box.propTypes = {
  ...space.propTypes,
  ...width.propTypes,
  ...color.propTypes,
  ...textAlign.propTypes
}

export default Box
