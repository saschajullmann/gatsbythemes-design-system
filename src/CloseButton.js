import React from 'react'
import IconButton from './IconButton'
import PropTypes from 'prop-types'
import theme from './theme'

const CloseButton = props => <IconButton {...props} name="close" />

CloseButton.defaultProps = {
  size: 24,
  title: 'close'
}

CloseButton.propTypes = {
  onClick: PropTypes.func,
  size: PropTypes.number,
  title: PropTypes.string
}

CloseButton.displayName = 'CloseButton'

export default CloseButton
