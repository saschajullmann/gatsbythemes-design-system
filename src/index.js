export { default as Absolute } from './Absolute'
export { default as BackgroundImage } from './BackgroundImage'
export { default as Badge } from './Badge'
export { default as Banner } from './Banner'
export { default as BlockLink } from './BlockLink'
export { default as Box } from './Box'
export { default as Button } from './Button'
export { default as Card } from './Card'
export { default as CloseButton } from './CloseButton'
export { default as Container } from './Container'
export { default as Divider } from './Divider'
export { default as Flag } from './Flag'
export { default as Flex } from './Flex'
export { default as GreenButton } from './GreenButton'
export { default as Heading } from './Heading'
export { default as Hide } from './Hide'
export { default as Hug } from './Hug'
export { default as Icon } from './Icon'
export { default as IconButton } from './IconButton'
export { default as IconField } from './IconField'
export { default as Image } from './Image'
export { default as Input } from './Input'
export { default as InputField } from './FormField'
export { default as FormField } from './FormField'
export { default as InputGroup } from './InputGroup'
export { default as Label } from './Label'
export { default as Link } from './Link'
export { default as OutlineButton } from './OutlineButton'
export { default as Radio } from './Radio'
export { default as RatingBadge } from './RatingBadge'
export { default as RedButton } from './RedButton'
export { default as Relative } from './Relative'
export { default as Stamp } from './Stamp'
export { default as Select } from './Select'
export { default as Text } from './Text'
export { default as Tooltip } from './Tooltip'
export { default as ToggleBadge } from './ToggleBadge'
export { default as Truncate } from './Truncate'
export { default as Checkbox } from './Checkbox'

export { default as theme } from './theme'
export { default as ThemeProvider } from './ThemeProvider'

export * from './theme'