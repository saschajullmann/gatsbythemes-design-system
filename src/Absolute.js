import styled from '@emotion/styled'
import Box from './Box'
import { top, right, bottom, left, zIndex } from 'styled-system'

const Absolute = styled(Box)`
  position: absolute;
  ${top} ${bottom} ${left} ${right}
  ${zIndex}
`

Absolute.propTypes = {
  ...top.propTypes,
  ...right.propTypes,
  ...bottom.propTypes,
  ...left.propTypes,
  ...zIndex.propTypes
}

Absolute.displayName = 'Absolute'

export default Absolute
