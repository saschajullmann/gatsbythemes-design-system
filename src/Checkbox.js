import React from 'react'
import styled from '@emotion/styled'
import { css } from '@emotion/core'
import PropTypes from 'prop-types'
import theme from './theme'
import Icon from './Icon'
import Box from './Box'

const Checkbox = props => {
  const { disabled, size } = props
  return (
    <CheckBoxWrapper disabled={disabled}>
      <StyledInput type="checkbox" {...props} />
      <Icon name="boxChecked" size={size} data-name="checked" />
      <Icon name="boxEmpty" size={size} data-name="empty" />
    </CheckBoxWrapper>
  )
}

const CheckBoxWrapper = styled(Box)`
  display: inline-block;
  position: relative;
  vertical-align: middle;
  cursor: pointer;
  color: ${props =>
    props.disabled ? props.theme.colors.borderGray : props.theme.colors.gray};

  svg[data-name='checked'] {
    display: none;
  }

  > input:checked {
    & ~ svg[data-name='checked'] {
      display: inline-block;
      color: ${props =>
    props.disabled
      ? props.theme.colors.borderGray
      : props.theme.colors.blue};
    }

    & ~ svg[data-name='empty'] {
      display: none;
    }
  }
`

CheckBoxWrapper.defaultProps = {
  theme
}

const StyledInput = styled.input`
  appearance: none;
  opacity: 0;
  position: absolute;
  z-index: 0;
`

Checkbox.displayName = 'Checkbox'

Checkbox.propTypes = {
  id: PropTypes.string.isRequired,
  size: PropTypes.number,
  onChange: PropTypes.func.isRequired
}

Checkbox.defaultProps = {
  size: 20
}

export default Checkbox
