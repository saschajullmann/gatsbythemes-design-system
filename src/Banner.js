import React from 'react'
import Box from './Box'
import Flex from './Flex'
import Text from './Text'
import Icon from './Icon'
import CloseButton from './CloseButton'
import Heading from './Heading'
import PropTypes from 'prop-types'

const bannerColors = {
  green: {
    backgroundColor: 'green',
    color: 'white',
    icon: 'success'
  },
  lightGreen: {
    backgroundColor: 'lightGreen',
    color: 'darkGreen',
    icon: 'success'
  },
  red: {
    backgroundColor: 'red',
    color: 'white',
    icon: 'warning'
  },
  lightRed: {
    backgroundColor: 'lightRed',
    color: 'darkRed',
    icon: 'warning'
  },
  orange: {
    backgroundColor: 'orange',
    color: 'white',
    icon: 'attention'
  },
  blue: {
    backgroundColor: 'blue',
    color: 'white',
    icon: 'information'
  },
  lightBlue: {
    backgroundColor: 'lightBlue',
    color: 'darkBlue',
    icon: 'information'
  }
}

const Banner = props => {
  const bannerColor = bannerColors[props.bg] || {}
  const icon = props.iconName || bannerColor.icon

  return (
    <Box
      {...props}
      bg={bannerColor.backgroundColor || props.bg}
      color={bannerColor.color || props.color}
    >
      <Flex justifyContent="space-between" alignItems="flex-start">
        {!!icon && !!props.showIcon && (
          <Icon name={icon} mr={2} size={24} mt="-2px" />
        )}
        <Box w={1}>
          <Text textAlign={props.textAlign}>
            <Heading.h5>{props.header}</Heading.h5>
            <Text.span fontSize={1}>{props.text}</Text.span>
            {props.children}
          </Text>
        </Box>
        {!!props.onClose && (
          <CloseButton
            onClick={props.onClose}
            ml={2}
            size={24}
            title="close"
            mt="-2px"
          />
        )}
      </Flex>
    </Box>
  )
}

Banner.displayName = 'Banner'

Banner.propTypes = {
  header: PropTypes.string,
  iconName: PropTypes.string,
  onClose: PropTypes.func,
  showIcon: PropTypes.bool,
  text: PropTypes.node,
  textAlign: PropTypes.string
}

Banner.defaultProps = {
  bg: 'green',
  textAlign: 'left',
  showIcon: true
}

export default Banner
