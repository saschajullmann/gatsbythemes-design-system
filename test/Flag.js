import React from 'react'
import renderer from 'react-test-renderer'
import { ThemeProvider, Flag } from '../src'

describe('Flag', () => {
  test('renders', () => {
    const json = renderer.create(<Flag />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with width prop', () => {
    const json = renderer.create(<Flag width={256} />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with hex value as bg color', () => {
    const json = renderer.create(<Flag width={256} bg="#085397" />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with theme color as bg color', () => {
    const json = renderer
      .create(
        <ThemeProvider>
          <Flag width={256} bg="purple" />
        </ThemeProvider>
      )
      .toJSON()
    expect(json).toMatchSnapshot()
  })
})
