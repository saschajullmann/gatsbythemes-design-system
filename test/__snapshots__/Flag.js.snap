// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flag renders 1`] = `
.c4 {
  margin-bottom: -8px;
  margin-right: -8px;
  width: 4px;
  color: #0a0;
}

.c6 {
  padding-left: 4px;
  padding-top: 4px;
  padding-bottom: 4px;
  color: #fff;
  background-color: #0a0;
}

.c8 {
  margin-left: -8px;
  width: 18px;
  color: #0a0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: 0px;
}

.c3 {
  width: 8px;
  height: 8px;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  background-image: linear-gradient(45deg,transparent 50%,rgba(0,0,0,0.5) 50%), linear-gradient(45deg,transparent 50%,green 50%);
  position: absolute;
  bottom: 0;
}

.c7 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  background-color: green;
  border-radius: 0 0;
  -webkit-transform: skew(-14deg);
  -ms-transform: skew(-14deg);
  transform: skew(-14deg);
  position: relative;
  z-index: 1;
}

.c5 {
  font-size: px;
  border-radius: 0 0 0;
  color: white;
  background-color: green;
  z-index: 2;
}

.c1 {
  position: relative;
}

@media screen and (min-width:32em) {
  .c6 {
    padding-left: 16px;
    padding-top: 8px;
    padding-bottom: 8px;
  }
}

@media screen and (min-width:32em) {
  .c0 {
    margin-left: -8px;
  }
}

@media screen and (max-width:31.99em) {
  .c2 {
    display: none;
  }
}

<div
  className="c0"
>
  <div
    className="c1 c2 "
  >
    <div
      className="c3 c4"
      color="green"
      width="4px"
    />
  </div>
  <div
    className="c5 c6"
    color="white"
  />
  <div
    className="c7 c8"
    color="green"
    width="18px"
  />
</div>
`;

exports[`Flag renders with hex value as bg color 1`] = `
.c4 {
  margin-bottom: -8px;
  margin-right: -8px;
  width: 4px;
  color: #085397;
}

.c6 {
  padding-left: 4px;
  padding-top: 4px;
  padding-bottom: 4px;
  color: #fff;
  background-color: #085397;
}

.c8 {
  margin-left: -8px;
  width: 18px;
  color: #085397;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: 0px;
  width: 256px;
}

.c3 {
  width: 8px;
  height: 8px;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  background-image: linear-gradient(45deg,transparent 50%,rgba(0,0,0,0.5) 50%), linear-gradient(45deg,transparent 50%,#085397 50%);
  position: absolute;
  bottom: 0;
}

.c7 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  background-color: #085397;
  border-radius: 0 0;
  -webkit-transform: skew(-14deg);
  -ms-transform: skew(-14deg);
  transform: skew(-14deg);
  position: relative;
  z-index: 1;
}

.c5 {
  font-size: px;
  border-radius: 0 0 0;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  color: white;
  background-color: #085397;
  z-index: 2;
}

.c1 {
  position: relative;
}

@media screen and (min-width:32em) {
  .c6 {
    padding-left: 16px;
    padding-top: 8px;
    padding-bottom: 8px;
  }
}

@media screen and (min-width:32em) {
  .c0 {
    margin-left: -8px;
  }
}

@media screen and (max-width:31.99em) {
  .c2 {
    display: none;
  }
}

<div
  className="c0"
  width={256}
>
  <div
    className="c1 c2 "
  >
    <div
      className="c3 c4"
      color="#085397"
      width="4px"
    />
  </div>
  <div
    className="c5 c6"
    color="white"
  />
  <div
    className="c7 c8"
    color="#085397"
    width="18px"
  />
</div>
`;

exports[`Flag renders with theme color as bg color 1`] = `
.c5 {
  margin-bottom: -8px;
  margin-right: -8px;
  width: 4px;
  color: #70b;
}

.c7 {
  padding-left: 4px;
  padding-top: 4px;
  padding-bottom: 4px;
  color: #fff;
  background-color: #70b;
}

.c9 {
  margin-left: -8px;
  width: 18px;
  color: #70b;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: 0px;
  width: 256px;
}

.c4 {
  width: 8px;
  height: 8px;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  background-image: linear-gradient(45deg,transparent 50%,#70b 50%);
  position: absolute;
  bottom: 0;
}

.c8 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  background-color: #70b;
  border-radius: 0 2px 2px 0;
  -webkit-transform: skew(-14deg);
  -ms-transform: skew(-14deg);
  transform: skew(-14deg);
  position: relative;
  z-index: 1;
}

.c6 {
  font-size: 12px;
  border-radius: 0 0 2px 0;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  color: #fff;
  background-color: #70b;
  z-index: 2;
}

.c2 {
  position: relative;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.5;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

@media screen and (min-width:32em) {
  .c7 {
    padding-left: 16px;
    padding-top: 8px;
    padding-bottom: 8px;
  }
}

@media screen and (min-width:32em) {
  .c1 {
    margin-left: -8px;
  }
}

@media screen and (max-width:31.99em) {
  .c3 {
    display: none;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
    width={256}
  >
    <div
      className="c2 c3 "
    >
      <div
        className="c4 c5"
        color="purple"
        width="4px"
      />
    </div>
    <div
      className="c6 c7"
      color="white"
    />
    <div
      className="c8 c9"
      color="purple"
      width="18px"
    />
  </div>
</div>
`;

exports[`Flag renders with width prop 1`] = `
.c4 {
  margin-bottom: -8px;
  margin-right: -8px;
  width: 4px;
  color: #0a0;
}

.c6 {
  padding-left: 4px;
  padding-top: 4px;
  padding-bottom: 4px;
  color: #fff;
  background-color: #0a0;
}

.c8 {
  margin-left: -8px;
  width: 18px;
  color: #0a0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: 0px;
  width: 256px;
}

.c3 {
  width: 8px;
  height: 8px;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  background-image: linear-gradient(45deg,transparent 50%,rgba(0,0,0,0.5) 50%), linear-gradient(45deg,transparent 50%,green 50%);
  position: absolute;
  bottom: 0;
}

.c7 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  background-color: green;
  border-radius: 0 0;
  -webkit-transform: skew(-14deg);
  -ms-transform: skew(-14deg);
  transform: skew(-14deg);
  position: relative;
  z-index: 1;
}

.c5 {
  font-size: px;
  border-radius: 0 0 0;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  color: white;
  background-color: green;
  z-index: 2;
}

.c1 {
  position: relative;
}

@media screen and (min-width:32em) {
  .c6 {
    padding-left: 16px;
    padding-top: 8px;
    padding-bottom: 8px;
  }
}

@media screen and (min-width:32em) {
  .c0 {
    margin-left: -8px;
  }
}

@media screen and (max-width:31.99em) {
  .c2 {
    display: none;
  }
}

<div
  className="c0"
  width={256}
>
  <div
    className="c1 c2 "
  >
    <div
      className="c3 c4"
      color="green"
      width="4px"
    />
  </div>
  <div
    className="c5 c6"
    color="white"
  />
  <div
    className="c7 c8"
    color="green"
    width="18px"
  />
</div>
`;
